geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -1637,8 +1640,13 @@ private void readMessages() {
               }
             }
             isHandShakeReader = true;
-            // Once we have read the handshake the reader can go away
-            break;
+            notifyHandshakeWaiter(false);
```

Comment:
handshakeRead and handshakeCancelled are only set to true in notifyHandshakeWaiter().  Why do we need to call that method again in this spot?

Reply:
updated